### PR TITLE
Fix taskbar icon grouping in Windows bundle (add `app_user_model_id` to bundle shortcut)

### PR DESCRIPTION
**Pull request by jaimergp**
# Fixes/Closes

Needed to consolidate changes in https://github.com/napari/packaging/pull/77. This culminates the fix for https://github.com/napari/packaging/issues/69, hopefully 🤞 

# Description

This adds the metadata necessary to set an [AppUserModelId](https://learn.microsoft.com/en-us/windows/win32/properties/props-system-appusermodel-id) on the Windows shortcut, which helps Windows handle things like grouping taskbar icons and windows.

# References

* https://learn.microsoft.com/en-us/windows/win32/properties/props-system-appusermodel-id
* https://github.com/conda/menuinst/pull/133
* https://github.com/conda/menuinst/issues/127
* https://github.com/spyder-ide/spyder/issues/20791#issuecomment-1505877877

## Type of change
- [X] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (changes required to run napari, tests, & CI smoothly)
- [ ] Enhancement (non-breaking improvements of an existing feature)
- [ ] Documentation (update of docstrings and deprecation information)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?
- [X] CI of upstream dependencies passes.
- [X] CI of installer creation + tests passes.
- [X] Manual checks with artifacts

## Final checklist:

- [X] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation - https://github.com/napari/docs/pull/202
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Comment by jaimergp** [comment]
I can confirm this is now functional in https://github.com/napari/packaging/pull/77.

File: resources/conda_menu_config.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
             "platforms": {
                 "win": {
                     "precommand": "set \"PYTHONHOME=\" & set \"PYTHONPATH=\"",
-                    "desktop": true
+                    "desktop": true,
+                    "app_user_model_id": "Napari.Napari"
```
Comment (jaimergp) [inline resources/conda_menu_config.json]:
```suggestion
                    "app_user_model_id": "napari.napari"
```

File: resources/conda_menu_config.json [hunk 2]
```diff
@@ -18,7 +18,8 @@
             "platforms": {
                 "win": {
                     "precommand": "set \"PYTHONHOME=\" & set \"PYTHONPATH=\"",
-                    "desktop": true
+                    "desktop": true,
+                    "app_user_model_id": "napari.napari"
```
Comment (jaimergp) [inline resources/conda_menu_config.json]:
```suggestion
                    "app_user_model_id": "napari.napari.viewer.__PKG_VERSION__"
```

Comment (jaimergp) [reply resources/conda_menu_config.json]: I don't know who is setting this internally (Qt?), but it uses the full version too, and they both need to match. Ideally the version wouldn't be set from the Qt side, but that's how it is now.

**Comment by Carreau** [comment]
I'm going to close and reopen to try to kick the label checking workflow.

**Review by Czaki — APPROVED** [review]

**Comment by Carreau** [comment]
I don't know why the checks are not running, is it because this does not touch any code file ?

**Comment by jaimergp** [comment]
Can we have this merged? :D Thanks!

**Comment by Czaki** [comment]
@jaimergp Could you adjust the title to describe the profit (as I understand it is for grouping icons on the status bar). Also, does this should be in 0.4.19 or 0.5.0?

**Comment by jaimergp** [comment]
> @jaimergp Could you adjust the title to describe the profit (as I understand it is for grouping icons on the status bar). Also, does this should be in 0.4.19 or 0.5.0?

Done! This should be available in whatever installer we build next. Do we need to port it to a hypothetical 0.4.19 branch?

**Pull request merged**